service/src/main/java/org/ehrbase/service/TemplateDBStorageService.java

```diff
@@ -51,23 +53,33 @@ public Set<String> findAllTemplateIds() {
         return I_TemplateStoreAccess.fetchAllTemplateIds(getDataAccess());
     }
 
+    @Override
+    public Optional<UUID> findUuidByTemplateId(String templateId) {
+        return getTemplateStoreAccessByTemplateId(templateId).getId();
+    }
+
     @Override
     public void storeTemplate(OPERATIONALTEMPLATE template, Short sysTenant) {
-        if (readOperationaltemplate(template.getTemplateId().getValue()).isPresent()) {
-            I_TemplateStoreAccess.getInstance(getDataAccess(), template, sysTenant)
-                    .update();
+        if (readOperationalTemplate(template.getTemplateId().getValue()).isPresent()) {
+            findUuidByTemplateId(TemplateUtils.getTemplateId(template))
+                    .ifPresent(uuid -> I_TemplateStoreAccess.getInstance(getDataAccess(), template, sysTenant, uuid)
```

Comment:
readOperationalTemplate is not necessary, use findUuidByTemplateId to begin with (+ ifPresentOrElse).